### PR TITLE
BL-714-fe-implement-heat-map-visualization

### PR DESCRIPTION
**Pull request by Mbravo36**
## Description

Plotly was used to create a reusable world heat map component, its appearance can update dynamically with data from the api, heat map contains total petitions and grant rates by year.

#### Video Link

https://www.loom.com/share/9f54faf8c22b4c739c0c6551d2cfeee4

#### Jira Link

https://bloomtechlabs.atlassian.net/browse/BL-714?atlOrigin=eyJpIjoiMDIzZWYyMzcwNWY3NDkxMDlmNTk1NmM3MGE5YjU2NjIiLCJwIjoiaiJ9

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by EstrellaDionis — APPROVED** [review]
Well implemented, the map itself looks better than what I personally imagined it to be. The video provided great details about what was happening as well, great work!

**Review by EricMarshallLambert — APPROVED** [review]

**Comment by EricMarshallLambert** [comment]
Lots of commits in this task. I think the scope of this ticket might have not been granular enough

File: src/components/pages/DataVisualizations/HeatMapContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+              : `Asylum Petitions In ${heatMapYear}`
+          }
+        />
+        <WorldHeatMap
+          yearSelection={heatMapYear}
+          reducerForm={reduceGrantRate}
+          apiData={heatMapData}
+          title={
+            heatMapYear === 'all'
+              ? 'Total Asylum Grant Rate % 2018-2021'
```
Comment (Joshmclemore) [inline src/components/pages/DataVisualizations/HeatMapContainer.jsx:85]: This is great. Once again, a very relevant set of data that doesn't overload the user (even on mobile)

File: src/components/pages/DataVisualizations/HeatMaps/WorldHeatMap.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          ),
+          text: apiData.map(region => region.location),
+          autocolorscale: true,
+        },
+      ]}
+      layout={{
+        title,
+        paper_bgcolor: '#f7e4ca',
+        hoverlabel: {
+          bordercolor: '#f7e4ca',
```
Comment (Joshmclemore) [inline src/components/pages/DataVisualizations/HeatMaps/WorldHeatMap.jsx:35]: This looks pretty simple to change if/when needed.

File: src/components/Layout/items.jsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// links are absolute and will need to be changed if implemented on the main site.
+// Links are absolute and will need to be changed if/when this app is add to the main site.
```
Comment (Joshmclemore) [inline src/components/Layout/items.jsx:1]: Great job keeping the comments useful and up to date!

**Review by Joshmclemore — APPROVED** [review]
Overall, great job. A much better display than "Page not found" and something that finally resembles a view that the stakeholders have been asking for, implemented on a design that they have approved. Looks like good progress!

**Review by Ricardorouge — APPROVED** [review]
I would've appreciated if you had shown exactly how the plotly used the information to generate the heatmap in the video, however I was able to figure it out from the code it seems pretty simple, the heat map design looks great overall and a great start for sure

**Review by shannonsullivan — APPROVED** [review]
Visualization looks great with nice styling. Comments are helpful and the code looks nice and clean. As commented above, there are a lot of commits as maybe the scope of this was too large. As mentioned, this is a reusable component, but it is a little confusing if this is part of Release 5 and we have not completed the preceding releases yet.

**Review by lucaselia10 — APPROVED** [review]

**Review by AlexiusThomas — COMMENTED** [review]
Looks great! Excellent walkthrough and detail, should be easy for anyone to use this and complete the process.

**Pull request merged**